src/varint.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+
+#include "common.h"
+#include "varint.h"
+
+uintmax_t decode_varint(const unsigned char **bufp)
```

Comment:
Yes, sorry, something along the lines of `git_varint_decode(size_t *buflen, const char *buf)` or some such.